core/src/main/scala/org/bitcoins/core/protocol/ln/LnTaggedFields.scala

```diff
@@ -141,46 +139,40 @@ object LnTaggedFields {
 
           loop(newRemaining, fields.:+(tag))
 
-        case _ :: _ | _ :: _ :: _ =>
-          throw new IllegalArgumentException("Failed to parse LnTaggedFields, needs 15bits of meta data to be able to parse")
         case Nil =>
           fields
+        case _ =>
```

Comment:
This is bad, the previous 
```scala

case _ :: _ | _ :: _ :: _ => ... 

```

Is creating pattern match exhaustiveness

```scala

case _ => 

```

is dangerous because `_` can be any type.